### ISSUE TITLE
setting class weights

### ISSUE THREAD

**Issue by nyk2001** [qubvel/segmentation_models.pytorch#495]
Hi All,

I am struggling with an issue. I am performing classification (binary). I am using a DiceLoss and it is working perfectly fine. Though, my background pixels are large in quantity compared to the positive class. 
- Is there a way I can apply weights for the positive class ?

So far, I am not able to figure out a way or find an example.

Any help will be really appreciated in this regard

**Comment by nyk2001** [comment]
@qubvel

**Comment by ShengtianSang** [comment]
> Hi All,
> 
> I am struggling with an issue. I am performing classification (binary). I am using a DiceLoss and it is working perfectly fine. Though, my background pixels are large in quantity compared to the positive class.
> 
> * Is there a way I can apply weights for the positive class ?
> 
> So far, I am not able to figure out a way or find an example.
> 
> Any help will be really appreciated in this regard

You can add the weights directly into your loss function, for example, if you're using the torch.nllloss as loss function, you could send a weight like w=[1.23, 2.02, 0.11] to initialize the torch.nllloss:    loss=torch.nllloss(weights=w).
